### PR TITLE
vesktop: add --enable-wayland-ime flag to NIXOS_OZONE_WL

### PR DESCRIPTION
**Pull request by LilleAila**
This pull request adds the `--enable-wayland-ime` flag to the `NIXOS_OZONE_WL` flags, so that IMEs work in vesktop on wayland.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ve/vesktop/package.nix [hunk 1]
```diff
@@ -128,6 +129,7 @@ stdenv.mkDerivation (finalAttrs: {
       makeWrapper ${electron}/bin/electron $out/bin/vesktop \
         --add-flags $out/opt/Vesktop/resources/app.asar \
         ${lib.optionalString withTTS "--add-flags \"--enable-speech-dispatcher\""} \
+        ${lib.optionalString withWaylandIME "--add-flags \"--enable-wayland-ime\""} \
         --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
```
Comment (Aleksanaa) [inline pkgs/by-name/ve/vesktop/package.nix:133]: Maybe add it here? Does `--enable-wayland-ime` bring any overhead to people not using an IME?

Comment (LilleAila) [reply pkgs/by-name/ve/vesktop/package.nix:133]: The value is set to false by default, so users who do not want it should not get any extra overhead.

**Review by getchoo — CHANGES_REQUESTED** [review]
https://github.com/NixOS/nixpkgs/pull/306310#discussion_r1576646996

**Comment by Aleksanaa** [comment]
> [#306310 (comment)](https://github.com/NixOS/nixpkgs/pull/306310#discussion_r1576646996)

small note: when you comment on others' reviews, you are already reviewing it

File: pkgs/by-name/ve/vesktop/package.nix (hunk 1 above)
Comment (LilleAila) [reply pkgs/by-name/ve/vesktop/package.nix]: I moved that flag to `NIXOS_OZONE_WL` and removed the option I created.

**Review by getchoo — APPROVED** [review]

**Review by vgskye — APPROVED** [review]
wait, IME didn't work with vesktop wayland? LGTM :3

**Comment by pluiedev** [comment]
> wait, IME didn't work with vesktop wayland? LGTM :3

It in fact didn't and that annoyed me to no end :] Had to keep doing `vesktop --enable-wayland-ime &; disown %1` manually

**Review by pluiedev — COMMENTED** [review]
Honestly... I prefer #294601's approach better where we unify what `NIXOS_OZONE_WL` does across all apps, including `--enable-wayland-ime`. This is okay as a stop-gap measure though

Also for future reference, perhaps the title and the description can be revised to reflect that no option has been added, and the PR is merely adding an additional flag

**Pull request merged**